auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BasicAuthToBearerTokenFilter.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth.http;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.BaseEncoding;
+import com.google.common.net.HttpHeaders;
+import com.palantir.tokens.auth.AuthHeader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link Filter} that replaces basic auth with a bearer token.
+ *
+ * <p>It assumes that the bearer token is held in the password field of the basic auth credentials, and that it is
+ * base-64 encoded.
+ */
+public class BasicAuthToBearerTokenFilter implements Filter {
+    private static final String BASIC_AUTH_STR = "Basic";
+    private static final Logger log = LoggerFactory.getLogger(BasicAuthToBearerTokenFilter.class);
```

Comment:
(nit) I prefer to put the logger in its own section b/c it's sort of a meta variable. like this:
```
private static final Logger log = ...;

private static final String BASIC_AUTH_STR = ...
private static final BaseEncoding BASE_64_ENCODING = ...
```